### ISSUE TITLE
Focus doesn't work properly on Safari 

### ISSUE THREAD

**Issue by GreedyA1** [angular/components#27500]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

the focus doesn't work the same on IOS/Android or Windows/MacOS
https://v11.material.angular.io/cdk/a11y/examples

Windows chrome/
https://github.com/angular/components/assets/18186211/853ee9e6-addf-4025-a1e0-05f071485fb3

Safari MacOS/
https://github.com/angular/components/assets/18186211/809cab96-9426-4bed-9b9c-0407b8efcc74

### Reproduction

https://stackblitz.com/angular/qoyljvexjjl?file=src%2Fapp%2Ffocus-monitor-directives-example.ts

### Expected Behavior

focus should change when clicked on Safari.

### Actual Behavior

focus doesn't change when clicked on Safari.

### Environment

- Angular:
- CDK/Material:
- Browser(s): Safari
- Operating: macOS

**Comment by wagnermaciel** [comment]
This issue description is also true for the latest version of Angular. I'll also mention that it's unclear if this is a real issue or just browser specific behavior

**Comment by crisbeto** [comment]
My understanding is that Safari just doesn't focus elements via click. For keyboard focus you have to enable it yourself https://stackoverflow.com/questions/1848390/safari-ignoring-tabindex. Closing since we can't do much about it.
